viewing_party/party.py

```diff
@@ -1,23 +1,186 @@
+
+        if genre not in counts:
+            # Add it into count dict
+            counts[genre] = 1
+        else:
+            # If already in count dict then increase its number
+            counts[genre] = counts[genre] + 1
+
+    highest_count = 0
+    highest_genre = None
+
+    for genre, count in counts.items():
+        # Compare if count value greater then the current highest
+        if count > highest_count:
+            # If true then set the highest count to be the count and set highest genre to genre
+            highest_count = count
+            highest_genre = genre
+
+    #print(highest_genre)
+    return highest_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    #Assign empty list to variables
+    user_unique_watched = []
+    friend_movie_title = []
+
+    #loop through wantched list from friend
+    for friend in user_data["friends"]:
+        friend_list = friend["watched"]
+        for movie in friend_list:
+            #add movie title into friend_movie_title list
+            friend_movie_title.append(movie["title"])
+
+    #loop through user's watched list 
+    for user in user_data["watched"]:
+        # if user's watched list movie not in friend's list then add it to the user_unique_watched list
+        if user["title"] not in friend_movie_title:
+            user_unique_watched.append(user)
+    return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    #Assign empty list to variables
+    user_movie_title = []
+    friend_unique_watched = []
+    f_uni_watched_rm_dup = []
+
+    for user in user_data["watched"]:
+        user_movie_title.append(user["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            #remove duplicate titles          
+            if movie["title"] not in user_movie_title \
+                and movie["title"] not in friend_unique_watched:
+                f_uni_watched_rm_dup.append(movie)
+                friend_unique_watched.append(movie["title"])
+
+    return f_uni_watched_rm_dup
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    friend_unique_movies = get_friends_unique_watched(user_data)
+
+    # Recommend only movies available on user’s subscriptions
+    recommended = [
+        movie for movie in friend_unique_movies
+        if movie["host"] in user_data["subscriptions"]
+    ]
+
+    #print(recommended)
+    return recommended
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    #grab user_movie_list, friend_movie_list, and user_genre
+    user_watched = get_unique_watched(user_data)

```

Comment:
Notice that we no longer need this call at all.

When removing one chunk of unnecessary code, keep an eye out for other areas that become unnecessary. VS Code helps us identify values that aren't used by showing the variable name more dimly than other variables, which is how I see `user_watched` displayed when opening this in VS Code.